Shared/Actions/CreateMenuItem.swift

```diff
@@ -0,0 +1,505 @@
+  """,
+		default: .default
+	)
+	var background: RMStyle
+	
+	// SF Symbol
+	@Parameter(
+		title: "SF Symbol Name",
+		description: """
+  The Name of a SF Symbol
+  
+  For available Symbols see Apple's Website (https://developer.apple.com/sf-symbols/) ])
+  """,
+		default: "plus"
+	)
+	var systemName: String
+	
+	@Parameter(
+		title: "Foreground",
+		description: "The Color for your SF Symbol",
+		default: .default
+	)
+	var foreground: RMStyle
+	
+	// Emoji
+	@Parameter(
+		title: "Emoji",
+		description: """
+  Any Emoji 😀.
+  
+  Tap the Emoji button on your keyboard and select one emoji.
+  """,
+		default: "😀",
+		inputOptions: .init(keyboardType: .default)
+	)
+	var emoji: String
+	
+	@Parameter(
+		title: "Data"
+	)
+	var data: String?
+	
+	@Parameter(
+		title: "Background Style",
+		description: """
+  The style of the icon's background.
+  """,
+		default: .circle
+	)
+	var backgroundShape: RMBackgroundShape
+	
+	func makeIcon() async -> Data? {
+		switch iconType {
+		case .sfSymbol:
+			return await RMIconContainer(
+				sfSymbol: systemName,
+				foregroundColor: foreground.color(),
+				backgroundColor: background.color(isBackground: true),
+				backgroundShape: backgroundShape
+			)
+			.render()
+		case .emoji:
+			return await RMIconContainer(
+				emoji: emoji,
+				backgroundColor: background.color(isBackground: true),
+				backgroundShape: backgroundShape
+			)
+			.render()
+		}
+	}
+	
+	func perform() async throws -> some IntentResult & ReturnsValue<MenuItem>  {
+		let icon = await makeIcon()
+		
+		let item = MenuItem(
+			title: menuTitle,
+			subtitle: subtitle,
+			icon: icon
+		)
+		
+		return .result(value: item)
+	}
+}
+
+// MARK: Menu Item
+struct MenuItem: TransientAppEntity {
+	init() {
+		self.init(title: nil, subtitle: nil)
+	}
+	
+	static let typeDisplayRepresentation: TypeDisplayRepresentation = "Menu Item"
+	
+	var displayRepresentation: DisplayRepresentation {
+		let title: LocalizedStringResource = "\(title ?? "")"
+		let subtitle: LocalizedStringResource? = if let subtitle  { "\(subtitle)" } else { nil }
+		let image: DisplayRepresentation.Image? = if let icon {
+			if #available(iOS 17.0, macOS 14.0, *) {
+				.init(data: icon.data, displayStyle: .default)
+			} else {
+				.init(data: icon.data)
+			}
+		} else {
+			nil
+		}
+		
+		return DisplayRepresentation(
+			title: title,
+			subtitle: subtitle,
+			image: image
+		)
+	}
+	
+	@Property(title: "Title")
+	var title: String?
+	
+	@Property(title: "Subtitle")
+	var subtitle: String?
+	
+	@Property(title: "Icon")
+	var icon: IntentFile?
+	
+	init(
+		title: String? = nil,
+		subtitle: String? = nil,
+		icon: Data? = nil
+	) {
+		if let icon {
+			self.icon = .init(data: icon, filename: "icon.png", type: .image)
+		} else {
+			self.icon = nil
+		}
+		self.title = title
+		self.subtitle = subtitle
+	}
+}
+
+// MARK: Style
+/**
+ A Style for an Icon or it's background
+ */
+enum RMStyle: String, AppEnum {
```

Comment:
Make types private whenever possible.

Reply:
Not possible there or anywhere else bc intent Metadata extraction will fail